### ISSUE TITLE
Embedded dashboard iframe missing title (accessibility)

### ISSUE THREAD

**Issue by bhaugeea** [apache/superset#27016]
## Screenshot

![image](https://github.com/apache/superset/assets/99675156/21b27456-0f69-4118-81e0-4c25412322e9)

## Description

Screen reader users may rely on the (missing) title attribute to know that the iframe is where the dashboard content is.

## Design input

Propose that the title be "Embedded Dashboard".
